andela/ah-the-answer-frontend
# #164829329 Ft implement comments crud

#### What does this PR do?
Implements CRUD functionality for comments 

#### Description of the task to be completed
- Enable authenticated users to post comments
- Enable users of comments while authenticated to edit or delete their comments
- Lists all available comments for a specific article
#### How should this be manually tested
- Clone the repo and navigate to the root directory using a terminal
- Install all the dependencies by typing `npm install`
- Start the server by typing `npm start`
- Navigate to the main page using a browser at http://localhost:3000/
<img width="1440" alt="Screenshot 2019-05-16 at 19 52 50" src="https://user-images.githubusercontent.com/26567020/57872505-7d659880-7814-11e9-9641-7b093a2a5204.png">

- Sign up and log in
<img width="1440" alt="Screenshot 2019-05-16 at 19 53 01" src="https://user-images.githubusercontent.com/26567020/57872556-9b32fd80-7814-11e9-9ecb-2b65cc7cb8fc.png">

- Once on the homepage click an article to view its details. You should see a comment form with a button labelled `comment`
<img width="1440" alt="Screenshot 2019-05-16 at 19 53 49" src="https://user-images.githubusercontent.com/26567020/57872584-a9811980-7814-11e9-9368-8d15286ac084.png">

- Write your comment in the form and press the button
- The comment should appear with an edit and delete button
<img width="1436" alt="Screenshot 2019-05-16 at 19 54 23" src="https://user-images.githubusercontent.com/26567020/57872604-b7369f00-7814-11e9-8af4-a2fc300bdfce.png">
 
- Click the edit button to edit the comment
- Click the delete button to delete the comment

#### Relevant PT story
https://www.pivotaltracker.com/story/show/164829329
